lxd/storage_buckets.go

```diff
@@ -55,6 +55,55 @@ var storagePoolBucketKeyCmd = APIEndpoint{
 	Put:    APIEndpointAction{Handler: storagePoolBucketKeyPut, AccessHandler: allowPermission(entity.TypeStorageBucket, auth.EntitlementCanEdit, "poolName", "bucketName")},
 }
 
+// storageBucketAccessHandler returns an access handler that checks for the given entitlement against a storage bucket.
+// The storage pool containing the bucket and the effective project of the bucket are added to the request context for
+// later use.
+func storageBucketAccessHandler(entitlement auth.Entitlement) func(d *Daemon, r *http.Request) response.Response {
```

Comment:
Why doesn't this function use an equivalent of addStoragePoolVolumeDetailsToRequestContext like storage volumes do?

Reply:
I've updated storage buckets to use the same pattern as for storage volumes.